### PR TITLE
changes to reflect dadi-cli updates

### PR DESCRIPTION
**Pull request by chriscrsmith**
I noticed the DFE side of the pipeline was broken, so I changed some command line flags and redirected some outputs. The pipeline runs for me now.

### CODE REVIEW

File: workflows/dfe.snake [hunk 1]
```diff
@@ -123,9 +123,9 @@ rule dadi_infer_dm:
     threads: 8
     shell:
         """
-        dadi-cli InferDM --fs {input} --model {params.demog} --p0 {params.demog_p0} --ubounds {params.demog_ubounds} --lbounds {params.demog_lbounds} --output-prefix {params.prefix} --optimizations {params.opts} --grids {params.grid_size} --threads {threads}
+        dadi-cli InferDM --fs {input} --model {params.demog} --p0 {params.demog_p0} --ubounds {params.demog_ubounds} --lbounds {params.demog_lbounds} --output-prefix {params.prefix} --optimizations {params.opts} --grids {params.grid_size} --cpus {threads} --nomisid
 
-        dadi-cli BestFit --input-prefix {params.prefix}.InferDM --model {params.demog} --ubounds {params.demog_ubounds} --lbounds {params.demog_lbounds}
+        dadi-cli BestFit --input-prefix {params.prefix}.InferDM --ubounds {params.demog_ubounds} --lbounds {params.demog_lbounds}
```
Comment (xin-huang) [inline workflows/dfe.snake:128]: `dadi-cli InferDM --fs {input} --model {params.demog} --p0 {params.demog_p0} --ubounds {params.demog_ubounds} --lbounds {params.demog_lbounds} --output-prefix {params.prefix} --optimizations {params.opts} --grids {params.grid_size} --cpus {threads} --nomisid --force-convergence`

The `BestFit` command can be replaced with the `force-convergence` option.

File: workflows/dfe.snake [hunk 2]
```diff
@@ -163,9 +163,9 @@ rule dadi_infer_dfe:
     threads: 8
     shell:
         """
-        dadi-cli InferDFE --fs {input[0]} --cache1d {input[1]} --demo-popt {input[2]} --output-prefix {params.prefix} --pdf1d {params.dfe} --p0 {params.dfe_p0} --ubounds {params.dfe_ubounds} --lbounds {params.dfe_lbounds} --ratio {params.ratio} --optimizations {params.opts} --threads {threads}
+        dadi-cli InferDFE --fs {input[0]} --cache1d {input[1]} --demo-popt {input[2]} --output-prefix {params.prefix} --pdf1d {params.dfe} --p0 {params.dfe_p0} --ubounds {params.dfe_ubounds} --lbounds {params.dfe_lbounds} --ratio {params.ratio} --optimizations {params.opts} --cpus {threads} --nomisid &>> {params.prefix}.log
 
-        dadi-cli BestFit --input-prefix {params.prefix}.InferDFE --pdf {params.dfe} --ubounds {params.dfe_ubounds} --lbounds {params.dfe_lbounds} --model {params.demog}
+        dadi-cli BestFit --input-prefix {params.prefix}.InferDFE --ubounds {params.dfe_ubounds} --lbounds {params.dfe_lbounds} &>> {params.prefix}.log
```
Comment (xin-huang) [inline workflows/dfe.snake:168]: `dadi-cli InferDFE --fs {input[0]} --cache1d {input[1]} --demo-popt {input[2]} --output-prefix {params.prefix} --pdf1d {params.dfe} --p0 {params.dfe_p0} --ubounds {params.dfe_ubounds} --lbounds {params.dfe_lbounds} --ratio {params.ratio} --optimizations {params.opts} --cpus {threads} --nomisid --force-convergence`

File: workflows/dfe.snake [hunk 3]
```diff
@@ -150,9 +150,9 @@ rule dadi_infer_dfe:
         output_dir + "/inference/{demog}/dadi/{dfes}/{annots}/{seeds}/{chrms}/pop{ids}/pop{ids}.two_epoch.spectra.bpkl",
         output_dir + "/inference/{demog}/dadi/{dfes}/{annots}/{seeds}/{chrms}/pop{ids}/pop{ids}.two_epoch.InferDM.bestfits"
     output:
-        output_dir + "/inference/{demog}/dadi/{dfes}/{annots}/{seeds}/{chrms}/pop{ids}/pop{ids}.two_epoch.gamma.InferDFE.bestfits"
+        output_dir + "/inference/{demog}/dadi/{dfes}/{annots}/{seeds}/{chrms}/pop{ids}/pop{ids}.two_epoch.gamma.InferDFE.bestfits",
+        output_dir + "/inference/{demog}/dadi/{dfes}/{annots}/{seeds}/{chrms}/pop{ids}/pop{ids}.two_epoch.gamma.log"
```
Comment (xin-huang) [inline workflows/dfe.snake:154]:
```suggestion
```

File: workflows/dfe.snake [hunk 4]
```diff
@@ -163,9 +163,9 @@ rule dadi_infer_dfe:
     threads: 8
     shell:
         """
-        dadi-cli InferDFE --fs {input[0]} --cache1d {input[1]} --demo-popt {input[2]} --output-prefix {params.prefix} --pdf1d {params.dfe} --p0 {params.dfe_p0} --ubounds {params.dfe_ubounds} --lbounds {params.dfe_lbounds} --ratio {params.ratio} --optimizations {params.opts} --threads {threads}
+        dadi-cli InferDFE --fs {input[0]} --cache1d {input[1]} --demo-popt {input[2]} --output-prefix {params.prefix} --pdf1d {params.dfe} --p0 {params.dfe_p0} --ubounds {params.dfe_ubounds} --lbounds {params.dfe_lbounds} --ratio {params.ratio} --optimizations {params.opts} --cpus {threads} --nomisid &>> {params.prefix}.log
```
Comment (xin-huang) [inline workflows/dfe.snake:166]:
```suggestion
        dadi-cli InferDFE --fs {input[0]} --cache1d {input[1]} --demo-popt {input[2]} --output-prefix {params.prefix} --pdf1d {params.dfe} --p0 {params.dfe_p0} --ubounds {params.dfe_ubounds} --lbounds {params.dfe_lbounds} --ratio {params.ratio} --optimizations {params.opts} --cpus {threads} --nomisid --force-convergence
```

File: workflows/dfe.snake (hunk 1 above)
Comment (xin-huang) [inline workflows/dfe.snake:128]:
```suggestion
```

Comment (xin-huang) [reply workflows/dfe.snake:128]: The `BestFit` command can be replaced with the `force-convergence` option in the `InferDM` and `InferDFE` commands.

File: workflows/dfe.snake (hunk 2 above)
Comment (xin-huang) [inline workflows/dfe.snake:168]:
```suggestion
```

**Comment by chriscrsmith** [comment]
Thanks for the edits @xin-huang. Pipeline runs fine for me

**Comment by andrewkern** [comment]
i went to review this PR, but ran into an issue with `dadi-cli` that I believe is unrelated. Here's that issue https://github.com/xin-huang/dadi-cli/issues/59

Hopefully that will be resolved quickly and we can merge this PR

**Comment by andrewkern** [comment]
okay that issue had to do with a dadi version problem. @chriscrsmith can you pin the version of dadi in the `environment.yml` file to `dadi==2.2.0` and push that commit to this PR? Then we can squash and merge

**Comment by chriscrsmith** [comment]
whoops! I messed something up.

**Comment by chriscrsmith** [comment]
also, I'm seeing a TON of files here: ./ext/dfe-alpha-release-2.16/data/n1_100_adv/n1_n2_t2_s_vec_dir/n1:100

are y'all getting the same?

**Comment by chriscrsmith** [comment]
right on. They aren't trivial size. But I'm fine with it if you are!

**Review by andrewkern — APPROVED** [review]

**Comment by chriscrsmith** [comment]
follow up: Andy reminded me those files are used by DFEalpha so they are supposed to be there

**Pull request merged**